src/libpmemfile-posix/file.c

```diff
@@ -1167,6 +1167,15 @@ _pmemfile_renameat2(PMEMfilepool *pfp,
 		goto end_unlock;
 	}
 
+	/*
+	 * From "rename" manpage:
+	 * "If oldpath and newpath are existing hard links referring to
+	 * the same file, then rename() does nothing, and returns a success
+	 * status."
+	 */
+	if (dst_vinode == src_vinode)
```

Comment:
Not that it matters a lot, but could this be checked earlier? While having the readlocks for their parents?